src/main/java/org/jabref/gui/mergeentries/MergeEntriesHelper.java

```diff
@@ -0,0 +1,101 @@
+
+package org.jabref.gui.mergeentries;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.jabref.gui.undo.NamedCompound;
+import org.jabref.gui.undo.UndoableChangeType;
+import org.jabref.gui.undo.UndoableFieldChange;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.FieldFactory;
+import org.jabref.model.entry.types.EntryType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for merging bibliography entries with undo support.
+ * Source entry data is merged into the library entry, with longer field values preferred
+ * and obsolete fields removed.
+ */
+public final class MergeEntriesHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MergeEntriesHelper.class);
+
+    private MergeEntriesHelper() {
+        // Private constructor to prevent instantiation
+    }
+
+    /**
+     * Merges two BibEntry objects with undo support.
+     * Following the typical source -> target pattern, but with domain-specific naming.
+     *
+     * @param entryFromLibrary The entry to be updated (target, from the library)
+     * @param source The entry containing new information (source, from the fetcher)
```

Comment:
Then, please name it `entryFromFetcher`. Speaking names! - Also rename in other helper methods.

Also swap arguments at the other methods!